src/arrayMethodSort.js

```diff
@@ -4,8 +4,23 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const compare =
+      compareFunction || ((a, b) => (String(a) > String(b) ? 1 : -1));
+
+    for (let i = 0; i < this.length - 1; i++) {
+      for (let j = 0; j < this.length - 1 - i; j++) {
+        if (compare(this[j], this[j + 1]) > 0) {
+          // Troca os elementos
+          const temp = this[j];
+
+          this[j] = this[j + 1];
+          this[j + 1] = temp;
```

Comment:
The swapping of elements is done correctly using a temporary variable. As an alternative, you could use array destructuring to swap the elements, which can make the code slightly more concise: [this[j], this[j + 1]] = [this[j + 1], this[j]];.